sql/show.go

```diff
@@ -20,11 +20,36 @@ package sql
 import (
 	"bytes"
 	"fmt"
+	"strings"
 
 	"github.com/cockroachdb/cockroach/sql/parser"
+	"github.com/cockroachdb/cockroach/util"
 	"github.com/cockroachdb/cockroach/util/encoding"
 )
 
+// Show a session-local variable name.
+func (p *planner) Show(n *parser.Show) (planNode, error) {
+	name := strings.ToUpper(n.Name)
+
+	v := &valuesNode{columns: []string{name}}
+
+	switch name {
+	case `DATABASE`:
+		v.rows = append(v.rows, []parser.Datum{parser.DString(p.session.Database)})
+	case `SYNTAX`:
+		switch parser.Syntax(p.session.Syntax) {
+		case parser.Modern:
+			v.rows = append(v.rows, []parser.Datum{parser.DString("modern")})
+		case parser.Traditional:
+			v.rows = append(v.rows, []parser.Datum{parser.DString("traditional")})
+		}
+	default:
+		return nil, util.Errorf("unknown variable: %s", name)
```

Comment:
I think this error needs a test. For errors returned to the user, we usually use `fmt.Errorf` as the line number isn't terribly useful to the end user.